### PR TITLE
Add GetAutomatedSecurityFixes to report status

### LINKED ISSUE #2080: Feature: Check if automated security fixes are enabled for a repository
In https://github.com/google/go-github/pull/1188 the endpoint was added to enable or disable automated security fixes.

However, there seems to be no way to check the current status of this setting. This is required to implement https://github.com/integrations/terraform-provider-github/issues/272.

I also noticed on the [REST api documentation page](https://docs.github.com/en/rest/reference/repos) there doesn't seem to be an API endpoint to check this (there is one for [checking vulnerability alerts](https://docs.github.com/en/rest/reference/repos#check-if-vulnerability-alerts-are-enabled-for-a-repository)).

Is it possible to implement this feature? If not, where do I create an issue for making such an endpoint available?

### PR DESCRIPTION
**Pull request by grahamhar**
This is an attempt to fix #2080. It is also required by upstream projects such as [terraform-provider-github](https://github.com/integrations/terraform-provider-github), see [issue 1301](https://github.com/integrations/terraform-provider-github/issues/1301)

```
go test github.com/google/go-github/...
ok      github.com/google/go-github/v53/github  1.165s
?       github.com/google/go-github/v53/test/fields     [no test files]
?       github.com/google/go-github/v53/test/integration        [no test files]

```

### CODE REVIEW

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:1240]:
```suggestion
	Enabled *bool `json:"enabled,omitempty"`
	Paused  *bool `json:"paused,omitempty"`
```

File: github/repos_test.go
Comment (gmlewis) [inline github/repos_test.go:690]:
```suggestion

	const methodName = "GetAutomatedSecurityFixes"
	testBadOptions(t, methodName, func() (err error) {
		_, _, err = client.Repositories.GetAutomatedSecurityFixes(ctx, "\n", "\n")
		return err
	})
	testNewRequestAndDoFailure(t, methodName, client, func() (*Response, error) {
		got, resp, err := client.Repositories.GetAutomatedSecurityFixes(ctx, "o", "r")
		if got != nil {
			t.Errorf("testNewRequestAndDoFailure %v = %#v, want nil", methodName, got)
		}
		return resp, err
	})
}
```

File: github/repos.go
Comment (grahamhar) [reply github/repos.go:1240]: @gmlewis could you explain this to me so I understand it's purpose in any future PR I do?

Comment (gmlewis) [reply github/repos.go]: Please let me know if https://github.com/google/go-github/pull/2827#issuecomment-1621990607 answers your question.

Comment (grahamhar) [reply github/repos.go]: Thanks for the link @gmlewis it makes it very clear. However, having read that I think my original form was possibly correct. Looking at the response schema in the API documentation it states that both fields are required in the response, so if I understood the link correctly these should not have omitempty set?

**Review by gmlewis — APPROVED** [review]
Thank you, @grahamhar !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

File: github/repos.go
Comment (gmlewis) [reply github/repos.go]: Ah, then I apologize.  If both are required, then you are right - not only does `omitempty` not need to be added, but these fields also do not need to be pointers.  They can simply be `bool`.

Comment (grahamhar) [reply github/repos.go]: let me update the PR

Comment (gmlewis) [reply github/repos.go]: Historically, we have erred on the side of using pointers since sometimes GitHub responses have included things in some API calls and omitted them in others.  But if these will always be present, then there is no need to make them pointers.  

However, I'm fine if you just want to drop the `omitempty` and revert to your original version.

Comment (grahamhar) [reply github/repos.go:1240]: Pushed the changes to remove the omitempty.

**Review by gmlewis — APPROVED** [review]

File: github/repos.go
Comment (vandanrohatgi) [inline github/repos.go:701]: Looks like API docs now recommend using the common `application/vnd.github+json` header. Maybe we can remove this custom header?

Comment (grahamhar) [reply github/repos.go:701]: Happy to do that as part of this PR.

Comment (grahamhar) [reply github/repos.go]: I've made the changes I think are needed please let me know if I need to do anything else.

**Review by vandanrohatgi — APPROVED** [review]
LGTM!

File: github/repos.go
Comment (vandanrohatgi) [reply github/repos.go]: Thank you @grahamhar !

**Comment by gmlewis** [comment]
Thank you, @vandanrohatgi !
Merging.

**Pull request merged**
